File: easy/189_Rotate_Array.cpp
#include "../libstruct.h"

using namespace std;

/*
 * No.189
 * Given an array, rotate the array to the right by k steps, where k is non-negative.
 */

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        if(nums.empty())
            return ;
        k %= nums.size();
        while(k)
        {
            int n = nums[nums.size() - 1];
            nums.pop_back();
            nums.insert(nums.begin(), n);
            k --;
        }
    }
};

File: easy/189_Rotate_Array.swift
//
//  189_Rotate_Array.swift
//  LeetCode
//
//  Created by wangzhe on 2018/7/16.
//  Copyright © 2018年 wangzhe. All rights reserved.
//

import Foundation

/*
 * Given an array, rotate the array to the right by k steps, where k is non-negative.
 */
class Solution {
    func rotate(_ nums: inout [Int], _ k: Int) {
        let slice = nums.count - k % nums.count
        let first: Array = nums[0..<slice].reversed()
        let second: Array = nums[slice..<nums.count].reversed()
        nums = first + second
        nums = Array(nums.reversed())
    }
}
